src/liballoc/collections/btree/map.rs

```diff
@@ -170,8 +171,13 @@ impl<K: Clone, V: Clone> Clone for BTreeMap<K, V> {
                 Internal(internal) => {
                     let mut out_tree = clone_subtree(internal.first_edge().descend());
 
+                    // Cannot call ensure_root_is_owned() because lacking K: Ord
```

Comment:
Then why not move that method to an impl without `K: Ord`?

Maybe that should also return `&mut Root`, like `Option::get_or_insert_with`, so you don't have to unwrap it immediately.

Reply:
Hm, for some reason I thought the bound was actually necessary, but I guess that doesn't make sense :)

I've done both of these now.